protocols/migration/migration_reports_4_0_0_to_reports_4_2_0.py

```diff
@@ -0,0 +1,237 @@
+from protocols import reports_4_0_0 as reports_4_0_0
+from protocols import reports_4_2_0
+from protocols.migration import BaseMigration
+from protocols.migration.participants import MigrationParticipants100To104
+
+
+class MigrateReports400To420(BaseMigration):
+
+    old_model = reports_4_0_0
+    new_model = reports_4_2_0
+
+    def migrate_cancer_interpretation_request(self, cancer_interpretation_request, sample_id):
+        """
+        :type cancer_interpretation_request: reports_4_0_0.CancerInterpretationRequest
+        :type sample_id: str
+        :rtype: reports_4_2_0.CancerInterpretationRequest
+        """
+        new_cancer_interpretation_request = self.new_model.CancerInterpretationRequest.fromJsonDict(
+            jsonDict=cancer_interpretation_request.toJsonDict()
+        )
+
+        new_cancer_interpretation_request.gitVersionControl = "4.2.0"
+
+        new_cancer_participant = MigrationParticipants100To104().migrate_cancer_participant(
+            cancer_participant=cancer_interpretation_request.cancerParticipant
+        )
+        new_cancer_interpretation_request.cancerParticipant = new_cancer_participant
+
+        # FIXME: we are using a sample_id received as a parameter, models need to be changed
+        tiered_variants = []
+        for reported_variant in cancer_interpretation_request.tieredVariants:
+            tiered_variants.append(self.migrate_reported_variant_cancer(
+                reported_variant,
+                sample_id
+            ))
+        new_cancer_interpretation_request.tieredVariants = tiered_variants
+
+        # FIXME: we have no sample id field in the structural variant
+        structural_tiered_variants = []
+        for reported_structural_variant in cancer_interpretation_request.structuralTieredVariants:
+            structural_tiered_variants.append(self.migrate_reported_structural_variant_cancer(
+                reported_structural_variant
+            ))
+        new_cancer_interpretation_request.structuralTieredVariants = structural_tiered_variants
+
+        return self.validate_object(
+            object_to_validate=new_cancer_interpretation_request, object_type=self.new_model.CancerInterpretationRequest
+        )
+
+    def migrate_cancer_interpreted_genome(self, cancer_interpreted_genome, sample_id):
+        """
+        :type cancer_interpreted_genome: reports_4_0_0.CancerInterpretedGenome
+        :type sample_id: str
+        :rtype: reports_4_2_0.CancerInterpretedGenome
+        """
+        new_cancer_interpreted_genome = self.new_model.CancerInterpretedGenome.fromJsonDict(
+            jsonDict=cancer_interpreted_genome.toJsonDict()
+        )
+
+        new_cancer_interpreted_genome.gitVersionControl = "4.2.0"
+
+        # FIXME: we are using the individual identifier in the cancer participant to set the sample id
```

Comment:
@priesgo do we need to change 4.2.0 before closing it ?

Reply:
no